app/repositories/PublishedRepository.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import javax.inject.{Inject, Singleton}
+
+import scala.concurrent.Future
+
+import play.api.libs.json.{Format, JsObject}
+import play.modules.reactivemongo.ReactiveMongoComponent
+
+import uk.gov.hmrc.mongo.ReactiveRepository
+
+import models.{PublishedProcess, RequestOutcome}
+import models.errors.{DatabaseError, Errors, NotFoundError}
+
+import repositories.formatters.PublishedProcessFormatter
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+//$COVERAGE-OFF$
+
+trait PublishedRepository {
+
+  def save(id: String, process: JsObject): Future[RequestOutcome[String]]
+  def getById(id: String): Future[RequestOutcome[JsObject]]
+}
+
+@Singleton
+class PublishedRepositoryImpl @Inject() (mongoComponent: ReactiveMongoComponent)
+    extends ReactiveRepository[PublishedProcess, String](
+      collectionName = "published",
```

Comment:
to be consistent with scratch, this would be publishedProcesses, although you renaming the scratchProcesses to scratch would also work.

Reply:
I will change the name of the collection as requested.